### PR TITLE
SPLICE-865 : Check if enterprise version is activated and if the user  try to use column privileges.

### PR DESCRIPTION
**Pull request by jfilali**
Straightforward, when the user try to do a GRANT on a column, we first check to see which version the user used and return a error message in case of the open source version.

### CODE REVIEW

File: db-shared/src/main/java/com/splicemachine/db/shared/common/reference/SQLState.java [hunk 1]
```diff
@@ -1986,6 +1986,7 @@ The message identifier (ie. the value that matches a field in this class)
 	String TABLE_NOT_PINNED				    						= "EXT32";
 	String DIRECTORY_REQUIRED			    						= "EXT33";
 	String ACCESSING_S3_EXCEPTION                                   = "EXT34";
+	String COLS_PERMS_EXCEPTION                                     = "EXT35";
```
Comment (jyuanca) [inline db-shared/src/main/java/com/splicemachine/db/shared/common/reference/SQLState.java]: EXT* is for external table related exception. You may want to put it close to authorization/authentication excpetions

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/GrantRevokeConstantOperation.java [hunk 2]
```diff
@@ -55,23 +59,38 @@ public void executeConstantAction( Activation activation ) throws StandardExcept
         LanguageConnectionContext lcc = activation.getLanguageConnectionContext();
         TransactionController tc = lcc.getTransactionExecute();
 
-		List <PermissionsDescriptor> permissionsDescriptors = privileges.executeGrantRevoke( activation, grant, grantees);
+        Manager manager = EngineDriver.driver().manager();
```
Comment (jyuanca) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/GrantRevokeConstantOperation.java]: Try to get ColPermsManager from Manager. If enterprise code is not enabled, an exception will be thrown, and you don't need to add another exception.

File: db-shared/src/main/java/com/splicemachine/db/shared/common/reference/SQLState.java (hunk 1 above)
Comment (jfilali) [inline db-shared/src/main/java/com/splicemachine/db/shared/common/reference/SQLState.java]: good catch

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/GrantRevokeConstantOperation.java [hunk 3]
```diff
@@ -55,23 +59,38 @@ public void executeConstantAction( Activation activation ) throws StandardExcept
...
+        if (privileges instanceof TablePrivilegeInfo) {
+            TablePrivilegeInfo info = (TablePrivilegeInfo) privileges;
+            if (!manager.isEnabled() && info.hasColumns()) {
+                throw StandardException.newException(SQLState.COLS_PERMS_EXCEPTION);
+            }
+        }
```
Comment (jyuanca) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/GrantRevokeConstantOperation.java]: If the column is not present, you should throw a different exception

Comment (jfilali) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/GrantRevokeConstantOperation.java]: this for perms only

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/GrantRevokeConstantOperation.java [hunk 4]
```diff
@@ -55,23 +59,38 @@ public void executeConstantAction( Activation activation ) throws StandardExcept
...
             return ProtoUtil.createRevokeSchemaPrivilege(txnId, schemaPermsDescriptor);
         }
         else if (permissionsDescriptor instanceof TablePermsDescriptor) {
             TablePermsDescriptor tablePermsDescriptor = (TablePermsDescriptor) permissionsDescriptor;
+
             return ProtoUtil.createRevokeTablePrivilege(txnId, tablePermsDescriptor);
         }
         else if (permissionsDescriptor instanceof ColPermsDescriptor) {
+
+            Manager manager = EngineDriver.driver().manager();
```
Comment (jyuanca) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/GrantRevokeConstantOperation.java]: Same problem here

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/GrantRevokeConstantOperation.java (hunk 2 above)
Comment (jfilali) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/GrantRevokeConstantOperation.java]: well no this need to happen only when the user use colum permission so i need to check myself

**Comment by jfilali** [comment]
@jyuanca  done, please review

**Comment by jfilali** [comment]
@snowspeeder  : one more +1

**Comment by jfilali** [comment]
@yxia92 : One more +1 ?

**Pull request merged**